internal/cmd/gocdk/internal/prompt/prompt.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prompt
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"gocloud.dev/internal/cmd/gocdk/internal/static"
+)
+
+// String prompts the user to input a string. It outputs msg to out, then
+// prompts the user to enter a value. If the user just hits return, the
+// value defaults to dflt.
+// The function only returns an error if the user enters "cancel".
+// TODO(rvangent): Add support for validation?
+// TODO(rvangent): Ctrl-C doesn't work; maybe replace cancel.
+func String(in io.Reader, out io.Writer, msg, dflt string) (string, error) {
+	fmt.Fprintln(out)
```

Comment:
This prompt seems verbose to me. IIUC the result is:

```
Enter a GCP region. default: us-central1
Enter a value, or "cancel":
```

What about:

```
Enter a GCP region [default: us-central1], or "cancel": 
```

(We could remove the `or "cancel"` if we support Ctrl-C, see above.)

Reply:
I was trying to follow what Terraform does. We can revisit how we want prompts to look later; added a TODO.